coverage-cop/action.yml

```diff
@@ -1,42 +1,123 @@
-      - name: Check coverage
         run: |
-          LINE_COVERAGE=$(lcov --list ${{ inputs.path }} | tail -n 1 | cut -d '|' -f 2 | sed -n "s/\([^%]*\)%.*/\1/p")
-          BRANCH_COVERAGE=$(lcov --rc lcov_branch_coverage=1 --list ${{ inputs.path }} | tail -n 1 | cut -d '|' -f 4 | sed -n "s/\([^%]*\)%.*/\1/p")
+          # ${{ env.stepName }}
+
+          # Print the received code cov report.
+          # TODO: The way it grabs the line/branch coverage is a little complicated
+          # I'd like to see if this can be done simpler.
+          echo -e " ${{ env.bashInfo }} Received LCov Report: ${{ inputs.line-coverage-min }} ${{ env.bashEnd}}"
+          lcov --list --rc lcov_branch_coverage=1 ${{ inputs.coverage-file }}
+          LINE_COVERAGE=$(lcov --list ${{ inputs.coverage-file }} | tail -n 1 | cut -d '|' -f 2 | sed -n "s/\([^%]*\)%.*/\1/p")
+          BRANCH_COVERAGE=$(lcov --rc lcov_branch_coverage=1 --list ${{ inputs.coverage-file }} | tail -n 1 | cut -d '|' -f 4 | sed -n "s/\([^%]*\)%.*/\1/p")
           RESULT=0
-          echo "Required line coverage: ${{ inputs.line-coverage-min }}"
-          echo "Line coverage:   $LINE_COVERAGE"
+
+          # Check Line Coverage
+          echo -e " ${{ env.bashInfo }} Required Line Coverage: ${{ inputs.line-coverage-min }} ${{ env.bashEnd}}"
+          echo -e " ${{ env.bashInfo }} Received Line Coverage: $LINE_COVERAGE ${{ env.bashEnd}}"
           if [[ $(echo "$LINE_COVERAGE < ${{ inputs.line-coverage-min }}" | bc) -ne 0 ]]; then
-            echo "Line Coverage is too low."
+            echo -e "${{ env.bashFail }} Line Coverage is too low. ${{ env.bashEnd }}"
             RESULT=1
           fi
-          echo "Required branch coverage: ${{ inputs.branch-coverage-min }}"
-          echo "Branch coverage: $BRANCH_COVERAGE"
+
+          echo -e " ${{ env.bashInfo }} Required Branch Coverage: ${{ inputs.branch-coverage-min }} ${{ env.bashEnd}}"
+          echo -e " ${{ env.bashInfo }} Received Branch Coverage: $BRANCH_COVERAGE ${{ env.bashEnd}}"
           if [[ $(echo "$BRANCH_COVERAGE < ${{ inputs.branch-coverage-min }}" | bc) -ne 0 ]]; then
-            echo "Branch Coverage is too low."
+            echo -e "${{ env.bashFail }} Branch Coverage is too low. ${{ env.bashEnd }}"
             RESULT=1
           fi
+
+          if [ $RESULT -eq 0 ]; then
+            echo -e "${{ env.bashPass }} ${{ env.stepName }} ${{ env.bashEnd }}"
+          else
+            echo "::group::Create Failed Codecov HTML Report"
+            genhtml --rc lcov_branch_coverage=1 --ignore-errors source ${{ inputs.coverage-file }} --legend --title "$(basename `git rev-parse --show-toplevel`) $(git rev-parse HEAD)" --output-directory=CodecovHTMLReport
+            zip -r CodecovHTMLReport.zip CodecovHTMLReport
+            echo "::endgroup::"
+            echo -e "${{ env.bashFail }} ${{ env.stepName }} ${{ env.bashEnd }}"
+          fi
           exit $RESULT
+
+      - name: Upload Failed Codecov HTML Report
```

Comment:
Why not upload the report even when it doesn't fail? This will reduce even further friction for developers and encourage them to squeeze out every possible test scenario possible.

Reply:
By default most repos require 100% line and branch coverage
Where felt un-neccessary to publish it every time?
But can easily do this by default if we want